internal/extproc/processor.go

```diff
@@ -81,9 +87,13 @@ func (p *Processor) ProcessRequestHeaders(_ context.Context, headers *corev3.Hea
 
 // ProcessRequestBody implements [Processor.ProcessRequestBody].
 func (p *Processor) ProcessRequestBody(ctx context.Context, rawBody *extprocv3.HttpBody) (res *extprocv3.ProcessingResponse, err error) {
+	p.requestStart = time.Now() // Track request metrics
+
 	path := p.requestHeaders[":path"]
 	model, body, err := p.config.bodyParser(path, rawBody)
 	if err != nil {
+		// Add to metrics tracker
+		metrics.RequestsTotal.WithLabelValues("unknown", "unknown", "error").Inc()
```

Comment:
I see that we're setting some metrics when there are errors, but not on all of them. Is it intentional? If not, this looks a bit error-prone, and easy to miss setting the error metric when adding new conditionals to the code int he future...

WDYT about refactoring the metrics functionality out to a processor that could wrap this one? something like an `InstrumentedProcessor` that wraps this one and emits the metrics when needed? This way it's easier to always emit the error metrics (when the inner processor returned an error), and don't miss that.
Also, the backend and model are published in the headers, so they should be accessible to the wrapping processor.

I don't know if given the current code and execution flows it is possible, but if it is, it would probably be cleaner

Reply:
The metrics is created in its own factory. When the extproc process starts, the metrics is there. If there is a misconstructed body, tracking it in the metrics is more helpful than dropping it silently or burry it in logs.

Reply:
+1

Reply:
@nacx i see what you meant. Yes, all errors should be tracked. The codebase is rapidly changing to catch up these errors :D I'll make sure the metrics are all the places you mentioned during next rebase.

Reply:
This is not resolved